varname/utils.py

```diff
@@ -277,6 +279,50 @@ def get_argument_sources(node: ast.Call,
     bound_args = signature.bind_partial(*arg_sources, **kwarg_sources)
     return bound_args.arguments
 
+def get_function_called_argname(
+        frame: FrameType,
+        node: ast.AST
+) -> Callable:
+    """Get the function who called argname"""
+    # We need node to be ast.Call
+    if not isinstance(node, ast.Call):
+        raise VarnameRetrievingError(
+            f"Expect an 'ast.Call' node, but got {type(node)!r}. "
+            "Are you using 'argname' inside a function?"
+        )
+
+    # variable
+    if isinstance(node.func, ast.Name):
+        return frame.f_locals.get(node.func.id,
+                                  frame.f_globals.get(node.func.id))
+
+    # use pure_eval
+    pure_eval_fail_msg = None
+    try:
+        from pure_eval import Evaluator, CannotEval
+    except ImportError:
+        pure_eval_fail_msg = "'pure_eval' is not installed."
+    else:
+        try:
+            evaluator = Evaluator.from_frame(frame)
+            return evaluator[node.func]
+        except CannotEval:
+            pure_eval_fail_msg = (
+                f"Cannot evaluate node {ast.dump(node.func)} "
+                "using 'pure_eval'."
+            )
+
+    # try eval
+    warnings.warn(
+        f"{pure_eval_fail_msg} "
+        "Using 'eval' to get the function that calls 'argname'. "
+        "Pass the function to 'argname' explicitly to get rid of this warning."
```

Comment:
This warning can have two audiences:

1. People who wrote code that directly uses argname, who can follow the advice above
2. People who use the code above, e.g. as a library, and can't change the call to argname. For them the advice should be to extract the function into a variable.